cluster-autoscaler/processors/provreq/injector.go

```diff
@@ -83,33 +92,39 @@ func (p *ProvisioningRequestPodsInjector) MarkAsFailed(pr *provreqwrapper.Provis
 func (p *ProvisioningRequestPodsInjector) GetPodsFromNextRequest(
 	isSupportedClass func(*provreqwrapper.ProvisioningRequest) bool,
 ) ([]*apiv1.Pod, error) {
+	if len(p.backoffDuration) >= p.maxCacheSize {
```

Comment:
nit: maybe add TODO to change it to LRU in the future?

Reply:
I have a TODO related to LRU on line 42, do you want to duplicate it here?

Reply:
I don't think it's quite the same thing. LRU is replacing the oldest element when the cache size would overflow, while a timeout just limits how long the element can spend in cache (whether it overflows or not).

Reply:
You're right. I think that's actually what we want, we don't want to remove a key after certain timeout, since we still retry ProvReq after maxBackoffTime. 
Added LRU cache and removed TODO.